chains/ethereum/listener.go

```diff
@@ -88,51 +152,43 @@ func buildQuery(contract ethcommon.Address, sig EventSig, startBlock *big.Int) e
 	return query
 }
 
+func buildSingleBlockQuery(contract ethcommon.Address, sig EventSig, startBlock *big.Int) eth.FilterQuery {
+	query := eth.FilterQuery{
+		FromBlock: startBlock,
+		ToBlock:   startBlock,
+		Addresses: []ethcommon.Address{contract},
+		Topics: [][]ethcommon.Hash{
+			{sig.GetTopic()},
+		},
+	}
+	return query
+}
```

Comment:
Please use the existing `buildQuery` method and add the `ToBlock` param to it. Since we're using `*big.Int` you can just pass `nil` if you don't want to specify it.